### ISSUE TITLE
[BUG] lombok.NonNull handled differently from checkerframework.NonNull

### ISSUE THREAD

**Issue by phxql** [rzwitserloot/lombok#2538]
**Describe the bug**

Hi. `lombok.NonNull` is handled differently from checkerframework.NonNull. lomboks `@NonNull` works on Constructors and method parameters. Checkerframeworks `@NonNull` only works on constructors.

**To Reproduce**

Given this class:

```java
import lombok.Value;
import org.checkerframework.checker.nullness.qual.NonNull;

@Value
class Test {
    @lombok.NonNull String lombokNonNull;
    @NonNull String checkerNonNull;

    public void lombokNonNull(@lombok.NonNull String foo) {
        System.out.println(foo);
    }

    public void checkerNonNull(@NonNull String foo) {
        System.out.println(foo);
    }
}
```

This

```java
new Test(null, "non-null");
```

fails with a NPE, expected and great. (testing lomboks `@NonNull` on ctor)

This

```java
new Test("non-null", null);
```

also fails with a NPE, which proves that lombok has some handling of checkerframeworks `@NonNull`. (testing checkerframeworks `@NonNull` on ctor)

However:

```java
new Test("", "").lombokNonNull(null);
```

fails with a NPE, expected and great. (testing lomboks `@NonNull` on methods)

But

```java
new Test("", "").checkerNonNull(null);
```

doesn't fail with a NPE. (testing checkerframeworks `@NonNull` on methods)

**Expected behavior**

I would expect that checkerframeworks `@NonNull` either works on the constructor and on the method or it doesn't work in either case. That it works on constructor but not on methods is strange.

It would be great if the checkerframework `@NonNull` is also applicable everywhere where lomboks `@NonNull` works.

**Version info (please complete the following information):**
- Lombok 1.18.12
- javac OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.7+10, mixed mode)

**Comment by Rawi01** [comment]
I agree that this behaviour is a little bit confusing. In general lombok only modifies/creates stuff that is annotated and  `checkerNonNull` is actually a normal method that works without lombok.

I think there are multiple solutions:
- Use [Accessors](https://projectlombok.org/features/experimental/Accessors) (this might work only in this example)
- Use both annotations at the same time
- Remove the generated `null` check for non-lombok `@NonNull` annotations. This will most likely break some stuff.
- Modify lombok to also add `null` checks for [all the other annotations](https://github.com/rzwitserloot/lombok/blob/7dfbe4323c15cbd88983380b27a250d2a381d148/src/core/lombok/core/handlers/HandlerUtil.java#L82). This might be even more confusing as it modifies classes without a single lombok annotation.
- Modify lombok to add `@NonNull` checker framework annotation if its own `@NonNull` annotation is present. Thats something thats already there for some other handlers.
- Use meta annotations (#557)

Disclaimer: I'm not a project owner.

**Comment by rspilker** [comment]
So far, our reasoning was: 
- If a field is annotated with any NonNull, the intention is 100% clear. We can add null-checks to constructors and setters.
- If we would insert null-checks for external NonNull annotations, we're not 100% certain that that's what expected.

**Comment by rspilker** [comment]
Currently, if you've configured you're using the checker framework in `lombok.config`, we generate a lot of checker framework annotations to help with static analysis. I'm not sure if the `org.checkerframework.checker.nullness.qual.NonNull` is one of them, but we could.
